src/components/ListItem.jsx

```diff
@@ -1,9 +1,83 @@
 import './ListItem.css';
+import { useEffect, useCallback } from 'react';
+import { updateItem } from '../api/firebase';
+
+const millisecondsIn24hrs = 86400000;
+
+export function ListItem({ listToken, item }) {
+	const currentDate = new Date();
+	const currentTime = Math.floor(currentDate.getTime());
+	let calcDate = item.dateLastPurchased
+		? item.dateLastPurchased.seconds * 1000
+		: null;
+
+	let timer = currentTime - calcDate;
```

Comment:
all these calculations should be happening inside the `useEffect` otherwise it will run everytime the component rerenders but we want it to run only in the initial render